Cargo.toml

```diff
@@ -24,3 +24,4 @@ byte = "0.2.4"
 websocket = "0.26.2"
 thiserror = "1.0"
 native-tls = "0.2.7"
+url = "2.2.2"
```

Comment:
Could we skip this dependency by using either `Reqwest`s or `websocket`s URL type? If not that's completely fine, I just wondered.

Reply:
Reqwest and websocket's URL types are a re-export of url's. This also allows us to have socket.rs not reference either Reqwest or Websocket directly (the individual transports should do all the communication with those libraries. 
https://github.com/seanmonstar/reqwest/blob/e6a1a09f0904e06de4ff1317278798c4ed28af66/src/lib.rs#L223
https://github.com/websockets-rs/rust-websocket/blob/b78d08b9bdb401a5ddfb5bea97240af7c49e9795/src/lib.rs#L56

Reply:
Ah I see, makes sense!